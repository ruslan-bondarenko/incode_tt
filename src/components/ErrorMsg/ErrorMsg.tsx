import { CloseCircleTwoTone } from "@ant-design/icons";
import { Flex, Typography } from "antd";
import React, { FC } from "react";

type Props = {
  error: string;
};

const ErrorMsg: FC<Props> = ({ error }) => {
  return (
    <Flex gap="middle" align="center" justify="center">
      <CloseCircleTwoTone twoToneColor="#ff4d4f" style={{ fontSize: "22px" }} />
      <Typography.Text type="danger">{error}</Typography.Text>
    </Flex>
  );
};

export default ErrorMsg;
