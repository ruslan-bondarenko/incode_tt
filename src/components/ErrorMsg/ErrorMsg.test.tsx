import { render, screen } from "@testing-library/react";
import React from "react";

import ErrorMsg from "./ErrorMsg";
import { describe, it } from "@jest/globals";

declare const expect: jest.Expect;

const error = "Server error!";

describe("ErrorMsg component", () => {
  it("ErrorMsg renders", () => {
    render(<ErrorMsg error={error} />);

    expect(screen.getByText(error)).toBeInTheDocument();
  });

  it("ErrorMsg snapshot", () => {
    const view = render(<ErrorMsg error={error} />);

    expect(view).toMatchSnapshot();
  });
});
