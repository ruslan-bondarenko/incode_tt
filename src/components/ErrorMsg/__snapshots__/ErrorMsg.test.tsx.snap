// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg component ErrorMsg snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-justify-center ant-flex-gap-middle"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle"
          role="img"
          style="font-size: 22px;"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              fill="#ff4d4f"
            />
            <path
              d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm171.8 527.1c1.2 1.5 1.9 3.3 1.9 5.2 0 4.5-3.6 8-8 8l-66-.3-99.3-118.4-99.3 118.5-66.1.3c-4.4 0-8-3.6-8-8 0-1.9.7-3.7 1.9-5.2L471 512.3l-130.1-155a8.32 8.32 0 01-1.9-5.2c0-4.5 3.6-8 8-8l66.1.3 99.3 118.4 99.4-118.5 66-.3c4.4 0 8 3.6 8 8 0 1.9-.6 3.8-1.8 5.2l-130.1 155 129.9 154.9z"
              fill="#fff2f0"
            />
            <path
              d="M685.8 352c0-4.4-3.6-8-8-8l-66 .3-99.4 118.5-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155-130.1 154.9a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3 99.3-118.5L611.7 680l66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.9 512.2l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
              fill="#ff4d4f"
            />
          </svg>
        </span>
        <span
          class="ant-typography ant-typography-danger css-dev-only-do-not-override-1kuana8"
        >
          Server error!
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-justify-center ant-flex-gap-middle"
    >
      <span
        aria-label="close-circle"
        class="anticon anticon-close-circle"
        role="img"
        style="font-size: 22px;"
      >
        <svg
          aria-hidden="true"
          data-icon="close-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            fill="#ff4d4f"
          />
          <path
            d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm171.8 527.1c1.2 1.5 1.9 3.3 1.9 5.2 0 4.5-3.6 8-8 8l-66-.3-99.3-118.4-99.3 118.5-66.1.3c-4.4 0-8-3.6-8-8 0-1.9.7-3.7 1.9-5.2L471 512.3l-130.1-155a8.32 8.32 0 01-1.9-5.2c0-4.5 3.6-8 8-8l66.1.3 99.3 118.4 99.4-118.5 66-.3c4.4 0 8 3.6 8 8 0 1.9-.6 3.8-1.8 5.2l-130.1 155 129.9 154.9z"
            fill="#fff2f0"
          />
          <path
            d="M685.8 352c0-4.4-3.6-8-8-8l-66 .3-99.4 118.5-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155-130.1 154.9a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3 99.3-118.5L611.7 680l66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.9 512.2l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
            fill="#ff4d4f"
          />
        </svg>
      </span>
      <span
        class="ant-typography ant-typography-danger css-dev-only-do-not-override-1kuana8"
      >
        Server error!
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
